flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogStateBackend.java

```diff
@@ -75,27 +77,37 @@
 
     private final StateBackend delegatedStateBackend;
 
+    private final boolean forSwitch;
```

Comment:
How about renamed to `restoreAndSwitchOnly`?

Naming is a hard thing.

Reply:
I agree, the name is not ver informative. 
But as I suggested above, maybe it's better to just split this class and get rid of the flag alltogether.

Reply:
I just split it into two classes.